src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -37,155 +40,283 @@ internal static class LibRdKafka
+
+                    if (!File.Exists(path))
+                    {
+                        dllDirectory = Path.Combine(
+                            baseDirectory, 
+                            is64 
+                                ? @"runtimes\win7-x64\native"
+                                : @"runtimes\win7-x86\native");
+                        path = Path.Combine(dllDirectory, "librdkafka.dll");
+                    }
+                }
+
+                if (WindowsNative.LoadLibraryEx(path, IntPtr.Zero, WindowsNative.LoadLibraryFlags.LOAD_WITH_ALTERED_SEARCH_PATH) == IntPtr.Zero)
+                {
+                    // catch the last win32 error by default and keep the associated default message
+                    var win32Exception = new Win32Exception();
+                    var additionalMessage =
+                        $"Error while loading librdkafka.dll or its dependencies from {path}. " +
+                        $"Check the directory exists, if not check your deployment process. " +
+                        $"You can also load the library and its dependencies by yourself " +
+                        $"before any call to Confluent.Kafka";
+
+                    throw new InvalidOperationException(additionalMessage, win32Exception);
+                }
+
+                isInitialized = SetDelegates(typeof(NativeMethods.NativeMethods));
+
+#else
+
+                const int RTLD_NOW = 2;
+
+                var nativeMethodTypes = new List<Type>();
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ||
+                    RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    if (userSpecifiedPath != null)
+                    {
+                        if (OsxNative.dlopen(userSpecifiedPath, RTLD_NOW) == IntPtr.Zero)
+                        {
+                            throw new DllNotFoundException($"Failed to load librdkafka: '{userSpecifiedPath}'");
+                        }
+                    }
+
+                    nativeMethodTypes.Add(typeof(NativeMethods.NativeMethods));
+                }
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    if (userSpecifiedPath != null)
+                    {
+                        if (LinuxNative.dlopen(userSpecifiedPath, RTLD_NOW) == IntPtr.Zero)
+                        {
+                            throw new DllNotFoundException($"Failed to load librdkafka: '{userSpecifiedPath}'");
+                        }
+                    
+                        nativeMethodTypes.Add(typeof(NativeMethods.NativeMethods));
+                    }
+                    else 
+                    {
+                        nativeMethodTypes.Add(typeof(NativeMethods.NativeMethods));
+                        nativeMethodTypes.Add(typeof(NativeMethods.NativeMethods_Debian9));
+                    }
+                }
```

Comment:
else  raise unsupported platform

Reply:
good idea. not a problem now, but may be in the future if more platforms are added.

Reply:
We want a meaningful error message (e.g., "Platform not supported") if people try to run this on an unsupported platform, regardless of which platforms we currently support.

Reply:
I mean we cover all platforms microsoft support, so the else will only get hit if microsoft start supporting additional platforms.